common/src/main/java/org/opensearch/ml/common/connector/HttpConnector.java

```diff
@@ -230,10 +242,14 @@ private void parseFromStream(StreamInput input) throws IOException {
         if (input.readBoolean()) {
             this.connectorClientConfig = new ConnectorClientConfig(input);
         }
+        if (streamInputVersion.onOrAfter(VERSION_2_19_0)) {
+            this.tenantId = input.readOptionalString();
+        }
```

Comment:
Does this work with BWC tests on 3.x branch?  I seem to remember having to commit with CURRENT and wait until 2.x backport to make this change and once it's on 2.19.0-SNAPSHOT changing it to this on main.

Reply:
Yeah I think it works. [Main](https://github.com/opensearch-project/ml-commons/blob/main/common/src/main/java/org/opensearch/ml/common/transport/connector/MLConnectorGetRequest.java#L45-L47) vs [2.x](https://github.com/opensearch-project/ml-commons/blob/2.x/common/src/main/java/org/opensearch/ml/common/transport/connector/MLConnectorGetRequest.java#L45-L47)

Reply:
Yeah it needs to _eventually_ be that way.

The issue is for BWC tests on this PR; if this change isn't merged to 2.x yet, it won't try to send/read that optional string on 2.19.0-SNAPSHOT, but a 3.0.0 node will detect that the stream _is_ 2.19.0 and try to send/read.  So rolling BWC will fail until it's backported.